src/core/service.c

```diff
@@ -1756,7 +1756,15 @@ static void service_enter_start(Service *s) {
         }
 
         if (!c) {
-                assert(s->type == SERVICE_ONESHOT);
+                if (s->type != SERVICE_ONESHOT) {
+                        /* There's no command line configured for the main command? Hmm, that is strange. This can only
+                         * happen if the configuration changes at runtime. In this case, let's enter a failure
+                         * state. */
+                        log_unit_error(UNIT(s), "There's no 'start' task anymore we could start: %m");
```

Comment:
Shouldn't this say "could not" ?  If it's been fixed in a later patch, I do apologise

Reply:
I think the message is correct.

Reply:
Oh, then maybe there should be 'that' between the anymore and we, alternatively the word 'for' at the end. Otherwise, and my english isn't the best, it can read slightly ambiguously.

Reply:
Could you open a new PR with your suggestion? I am not native...